Problem building LAMMPs with the QUIP package

Hi!

I’m trying to compile LAMMPs with the QUIP package using the Intel compilers.
However, the compilation fails when trying to link the CXX executable lmp with the error:
`ld: liblammps.so.0: undefined reference to 'lb2_'`

I’m using a libquip.a also build with the intel compilers using thelinux_x86_64_ifort_icc makefile.
Something worth noting is that compiling LAMMPs without QUIP using the Intel compilers works and compiling LAMMPs with QUIP but using the gcc compilers also works.
I have tracked down the error to the fact that LAMMPs is compiled with shared libs, this seems to cause a problem when including the QUIP package.
But since everting works fine using the gcc compilers is suspect that the error comes from the compilation of libquip.a
Are the any compiler flags that I need to modify in the Makefile.linux_x86_64_ifort_icc file in order to create a libquip.a that can be used to build LAMMPS with QUIP and shared libs using the Intel compilers?

I attaced the logfile with the error
[easybuild-LAMMPS-29Oct2020-20210823.153124.ChdBQ.log](https://github.com/libAtoms/QUIP/files/7036428/easybuild-LAMMPS-29Oct2020-20210823.153124.ChdBQ.log)


/Daniel

Intermediate feedback: I have trouble replicating the original error. Instead I get
```
liblammps.so.0: undefined reference to `quip_lammps_api_version'
liblammps.so.0: undefined reference to `quip_lammps_wrapper'
liblammps.so.0: undefined reference to `quip_lammps_potential_initialise'
```
for the `stable_23Jun2022_update2` version with `ifort (IFORT) 2021.8.0 20221119` and the current public of QUIP/GAP (`5ce3d4147`, `926af0c`) using the following configuration
```
cmake ../cmake -D BUILD_SHARED_LIBS=yes -D PKG_ML-QUIP=yes -D QUIP_LIBRARY=$QUIP_ROOT/build/$QUIP_ARCH/libatoms.a -D CMAKE_C_COMPILER=icc -D CMAKE_CXX_COMPILER=icpc -D CMAKE_Fortran_COMPILER=ifort
```

@RemiLacroix-IDRIS Which versions and configuration have you used to test this?

> @Sideboard : I used the same version of LAMMPS but I let it download and compile QUIP for me.

And which modules do you load / packages do you activate / options do you use? As of now, the auto build of QUIP fails for me:
```
cmake ../cmake -D BUILD_SHARED_LIBS=yes -D PKG_ML-QUIP=yes -D CMAKE_C_COMPILER=icc -D CMAKE_CXX_COMPILER=icpc -D CMAKE_Fortran_COMPILER=ifort
⋮
cmake --build .
⋮
ar: error.o: No such file or directory
make[4]: *** [Makefile:131: libatoms.a] Error 1
make[3]: *** [Makefile:155: libAtoms] Error 2
gmake[2]: *** [CMakeFiles/quip_build.dir/build.make:86: quip_build-prefix/src/quip_build-stamp/quip_build-build] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:762: CMakeFiles/quip_build.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

--- a/src/libAtoms/minimization.f95
+++ b/src/libAtoms/minimization.f95
@@ -102,7 +102,7 @@ module minimization_module
   end interface
 
   ! LBFGS stuff
-  external LB2
+  external QUIPLB2
   integer::MP,LP
   real(dp)::GTOL,STPMIN,STPMAX
   common /lb3/MP,LP,GTOL,STPMIN,STPMAX

